flytekit/types/structured/structured_dataset.py

```diff
@@ -569,7 +570,19 @@ def encode(
         sd_model.metadata._structured_dataset_type.format = handler.supported_format
         return Literal(scalar=Scalar(structured_dataset=sd_model))
 
+    @overload
     def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type: Type[T]) -> T:
+        ...
+
+    @overload
```

Comment:
Why do we need to add `to_python_value`? we have three `to_python_value`...

Reply:
i was doing this function type overloading https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading as mypy was complaining when I had union on both param and return value so it couldn't map the correct ones.

However, i removed it now in latest changes and it seemed to be fine running locally - so ill wait for ci to see if it passes.